data-prepper-core/src/main/java/org/opensearch/dataprepper/plugin/ComponentPluginArgumentsContext.java

```diff
@@ -81,20 +78,23 @@ private ComponentPluginArgumentsContext(final Builder builder) {
     }
 
     @Override
-    public Object[] createArguments(final Class<?>[] parameterTypes, Optional<Object> optionalArgument) {
-        if (optionalArgument.isPresent()) {
-            if (optionalArgument.get() instanceof ByteDecoder) {
-                ByteDecoder byteDecoder = (ByteDecoder) optionalArgument.get();
+    public Object[] createArguments(final Class<?>[] parameterTypes, final Object ... args) {
+        Map<Class<?>, Supplier<Object>> optionalArgumentsSuppliers = new HashMap<>();
+        Boolean byteDecoderInitialized = false;
```

Comment:
The plugin framework shouldn't know or care about `ByteDecoder` at all. Any special code for that should be in `PipelineParser`.

Also, I'm not sure what this condition is really doing other than possibly avoiding passing the parameter in twice.

I think this code can be:

```
Map<Class<?>, Supplier<Object>> optionalArgumentsSuppliers = new HashMap<>();
for(final Object arg: args) {
  optionalArgumentsSuppliers.put(arg.getClass(), () -> arg);
}
```